tests/v2/validation/charts/backup_restore/backup_restore.go

```diff
@@ -0,0 +1,321 @@
+	if err != nil {
+		return err
+	}
+
+	config.LoadConfig(charts.BackupRestoreConfigurationFileKey, backupRestoreConfig)
+
+	chartInstallOptions := &charts.InstallOptions{
+		Cluster:   cluster,
+		Version:   latestBackupVersion,
+		ProjectID: project.ID,
+	}
+	chartFeatureOptions := &charts.RancherBackupOpts{
+		VolumeName:                backupRestoreConfig.VolumeName,
+		BucketName:                backupRestoreConfig.S3BucketName,
+		CredentialSecretName:      secretName,
+		CredentialSecretNamespace: backupRestoreConfig.CredentialSecretNamespace,
+		Enabled:                   backupEnabled,
+		Endpoint:                  backupRestoreConfig.S3Endpoint,
+		Folder:                    backupRestoreConfig.S3FolderName,
+		Region:                    backupRestoreConfig.S3Region,
+	}
+
+	_, err = createOpaqueS3Secret(client.Steve)
+	if err != nil {
+		return err
+	}
+
+	logrus.Info("Installing backup chart")
+	err = charts.InstallRancherBackupChart(client, chartInstallOptions, chartFeatureOptions, true)
+	if err != nil {
+		return err
+	}
+
+	logrus.Info("Waiting for backup chart deployments to have expected number of available replicas")
+	err = charts.WatchAndWaitDeployments(client, project.ClusterID, backupChartNamespace, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+func createOpaqueS3Secret(steveClient *v1.Client) (string, error) {
+	config.LoadConfig(charts.BackupRestoreConfigurationFileKey, backupRestoreConfig)
+
+	logrus.Infof("Creating an opaque secret with name: %v", secretName)
+	secretTemplate := secrets.NewSecretTemplate(secretName, backupRestoreConfig.CredentialSecretNamespace, map[string][]byte{"accessKey": []byte(backupRestoreConfig.AccessKey), "secretKey": []byte(backupRestoreConfig.SecretKey)}, corev1.SecretTypeOpaque)
+	createdSecret, err := steveClient.SteveType(secrets.SecretSteveType).Create(secretTemplate)
+
+	return createdSecret.Name, err
+}
+
+func createProject(client *rancher.Client, clusterID string) (*management.Project, error) {
+	createProject, _, err := projectsapi.CreateProjectAndNamespace(client, clusterID)
+	if err != nil {
+		return nil, err
+	}
+
+	return createProject, nil
+}
+
+func createRole(client *rancher.Client, context string, roleName string, rules []management.PolicyRule) (role *management.RoleTemplate, err error) {
+	role, err = client.Management.RoleTemplate.Create(
+		&management.RoleTemplate{
+			Context: context,
+			Name:    roleName,
+			Rules:   rules,
+		})
+
+	return
+}
+
+func createRancherResources(client *rancher.Client, resourceCount int, clusterID string) ([]*management.User, []*management.Project, []*management.RoleTemplate, error) {
+	userList := []*management.User{}
+	projList := []*management.Project{}
+	roleList := []*management.RoleTemplate{}
+
+	for i := 0; i < resourceCount; i++ {
+		u, err := users.CreateUserWithRole(client, users.UserConfig(), globalUserRole)
+		if err != nil {
+			return userList, projList, roleList, err
```

Comment:
curious what this looks like when it runs; it would appear that this first if statement would always hit, returning empty lists.